src/modify.js

```diff
@@ -66,12 +101,122 @@ function rewriteAllFields(schema, configCallback, context) {
   });
 }
 
+function pickFields(originalSchema, pickConfig) {
+  if (!pickConfig) return originalSchema;
+
+  const fieldsToPick = pickConfig.fields;
+  const newSchema = {
+    properties: {},
+  };
+
+  Object.entries(originalSchema).forEach(([attrKey, attrValue]) => {
+    switch (attrKey) {
+      case 'properties':
+        // TODO — handle recursive nested fieldsets
+        fieldsToPick.forEach((fieldPath) => {
+          set(newSchema.properties, fieldPath, attrValue[fieldPath]);
+        });
+        break;
+      case 'x-jsf-order':
+      case 'required':
+        newSchema[attrKey] = attrValue.filter((fieldName) => fieldsToPick.includes(fieldName));
+        break;
+      case 'allOf': {
+        const newAllOf = [];
+        // remove conditional ("if, then, else") if it does not contain any reference to fieldsToPick
+        originalSchema.allOf.forEach((condition) => {
+          if (isConditionalReferencingAnyPickedField(condition, fieldsToPick)) {
+            newAllOf.push(condition);
+          }
+        });
+        newSchema[attrKey] = newAllOf;
+
+        break;
+      }
+      default:
+        newSchema[attrKey] = attrValue;
+    }
+  });
+
+  // Look for unpicked fields in the conditionals...
+  let missingFields = {};
+
+  newSchema.allOf.forEach((condition, ix) => {
+    const { if: ifCondition, then: thenCondition, else: elseCondition } = condition;
+
+    missingFields = {
+      ...missingFields,
+      ...findMissingFields(ifCondition, {
+        fields: fieldsToPick,
+        path: `allOf[${ix}].if`,
+      }),
+      ...findMissingFields(thenCondition, {
+        fields: fieldsToPick,
+        path: `allOf[${ix}].then`,
+      }),
+      ...findMissingFields(elseCondition, {
+        fields: fieldsToPick,
+        path: `allOf[${ix}].else`,
+      }),
+    };
+  });
+
+  if (Object.keys(missingFields).length > 0) {
+    // Read them to the new schema...
+    Object.entries(missingFields).forEach(([fieldName]) => {
+      set(newSchema.properties, fieldName, originalSchema.properties[fieldName]);
+    });
+    // And warn about it (the most important part!)
+    pickConfig.onWarn({
+      message: 'You picked a field with conditional fields. They got added:',
+      missingFields,
+    });
+  }
+
+  return newSchema;
+}
+
+function findMissingFields(conditional, { fields, path }) {
+  if (!conditional) {
+    return null;
+  }
+
+  let missingFields = {};
+
+  conditional.required?.forEach((fieldName) => {
+    if (!fields.includes(fieldName)) {
+      missingFields[fieldName] = {
+        path,
+      };
+    }
+  });
+
+  Object.entries(conditional.properties || []).forEach(([fieldName, fieldAttrs]) => {
+    if (!fields.includes(fieldName)) {
+      missingFields[fieldName] = { path };
+    }
+
+    if (fieldAttrs.properties) {
+      const nested = findMissingFields(fieldAttrs.properties, fields);
```

Comment:
Is this correct? Doesn't `findMissingFields` expect a conditional? I'm a bit lost by what `fieldAttrs.properties` refers to in this context since we're already going through `conditional.properties` above.

Reply:
Let me get back to you first thing tomorrow morning to clarify this and apply all the other changes

Reply:
Oh this one is to be recursive (\*doub-checked if word is well written\*), and nested fields, for example:

```js
{
  if: {
    properties: {
      address: {
        properties: {
          number: {
            const: 5
          }
        }
      }
    }
  },
  then: { /*...*/ },
  else: { /* ... */ }
}
```

I'll add a test in 1h after my call now.

Reply:
Actually, supporting conditionals with nested fields (fieldsets) has its own edge-cases too: It also needs to take into account the nested `allOf` (eg `properties.address.allOf`).

As I don't know of any case needing that right now, and I'm short on time, I rather leave it out for another day.

 I added more details to the current TODOs in the tests files.

In other words, I completely removed this block of code.

Reply:
Sounds good, thank you!